app/models/settings/base.rb

```diff
@@ -0,0 +1,166 @@
+# This was taken from the rails-settings-cached gem and adapted to work with
+# more than one settings model by making the request cache class-specific. We
+# also removed features we don't need like cache scopes and readonly fields.
+#
+# See: https://github.com/huacnlee/rails-settings-cached
+module Settings
+  # rubocop:disable Security/YAMLLoad
+  class Base < ActiveRecord::Base # rubocop:disable Rails/ApplicationRecord
+    self.abstract_class = true
+
+    class SettingNotFound < RuntimeError; end
+
+    class ProcetedKeyError < ArgumentError; end
+
+    PROTECTED_KEYS = %w[var value].freeze
+    SEPARATOR_REGEXP = /[\n,;]+/.freeze
+
+    class << self
+      def inherited(subclass)
+        super
+
+        # This defines a custom request cache based on ActiveSupport::CurrentAttributes.
```

Comment:
This is the old monkey-patch but written in a cleaner and more understandable way. The comment hopefully provides enough context,